syne_tune/optimizer/schedulers/botorch/botorch_searcher.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from typing import Dict, Optional, List
+import logging
+
+import numpy as np
+import torch
+from botorch.models import SingleTaskGP
+from botorch.fit import fit_gpytorch_model
+from botorch.models.transforms import Warp
+from botorch.utils import standardize
+from botorch.utils.transforms import normalize
+from gpytorch.mlls import ExactMarginalLogLikelihood
+from botorch.acquisition import qExpectedImprovement
+from botorch.optim import optimize_acqf
+from gpytorch.utils.errors import NotPSDError
+
+import syne_tune.config_space as cs
+from syne_tune.optimizer.schedulers.searchers import SearcherWithRandomSeed
+
+from syne_tune.optimizer.schedulers.searchers.bayesopt.datatypes.hp_ranges_factory import make_hyperparameter_ranges
+
+
+logger = logging.getLogger(__name__)
+
+
+class BotorchSearcher(SearcherWithRandomSeed):
+
+    def __init__(
+            self,
+            config_space: Dict,
+            metric: str,
+            num_init_random_draws: int = 3,
+            mode: str = "min",
+            points_to_evaluate: Optional[List[Dict]] = None,
+            fantasising: bool = True,
+            max_num_observations: Optional[int] = 200,
+            input_warping: bool = True,
+            **kwargs,
+    ):
+        """
+        A searcher that suggest configurations using BOTORCH to build GP surrogate and optimize acquisition functions.
+        :param config_space: configuration space to optimize
+        :param metric: metric to optimize, should be present in reported results.
+        :param num_init_random_draws: number of initial random draw, after this number the suggestion are obtained
```

Comment:
draw -> draws ;  maybe also add: "... the suggestions are obtained from the GP model"